sim-lib/src/lnd.rs

```diff
@@ -60,7 +62,27 @@ impl LightningNode for LndNode {
         unimplemented!()
     }
 
-    async fn track_payment(&self, _hash: PaymentHash) -> Result<(), LightningError> {
-        unimplemented!()
+    async fn track_payment(&self, hash: PaymentHash) -> Result<PaymentOutcome, LightningError> {
+        let mut client = self.client.clone();
+        let router_client = client.router();
+
+        let response = router_client
+            .track_payment_v2(TrackPaymentRequest {
+                payment_hash: hash.0.to_vec(),
+                no_inflight_updates: false,
+            })
+            .await?;
+
+        let mut stream = response.into_inner();
+
+        while let Some(payment) = stream.message().await? {
+            info!("Payment: {payment:?}");
+            if payment.status == 2 {
```

Comment:
Does `tonic_lnd` define an enum for the status? Would make this a little more readable